cli/lib/kontena/cli/stacks/upgrade_command.rb

```diff
@@ -83,20 +82,30 @@ def process_data(old_data, new_data)
       new_data.reverse_each do |stackname, data|
         reader = data[:loader].reader
         set_env_variables(stackname, current_grid) # set envs for execution time
```

Comment:
Best to keep the `set_env_variables`  as close to the `reader.execute` as possible. Probably doesn't matter for the `values.merge` bit, but still, don't want this drifting any further.